### PR TITLE
Add correct return to single selection from MultiNodeEdit

### LINKED ISSUE #49451: Inspector shows MultiNodeEdit when only one node is selected after deselecting
### Godot version

3.3.stable 3.x(f3d931410) master(090e4c4abd)

### System information

macOS 11.4

### Issue description

I selected two nodes in the editor and then decided to only change one of them, so I deselected the other one. But changing properties in the Inspector changes both nodes.

The inspector still shows `MultiNodeEdit` after the deselection.

Instead of CTRL-clicking to deselect one node, you can also click one of the nodes in the Scene dock. In this case, clicking the second node works fine, but clicking the first node still shows `MultiNodeEdit`.

### Steps to reproduce

1. Create an empty scene
2. Add two nodes (e.g. drag two icon.png into the scene)
3. Select both of them
4. Select only one of them by CTRL/CMD click the other one
5. Now the Inspector should still say `MultiNodeEdit`, and changing any property would affect both the nodes

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by ghost**
Fixes #49451 
deepin-screen-recorder_Godot_Engine_20210617111915

### CODE REVIEW

File: editor/scene_tree_dock.cpp [hunk 1]
```diff
@@ -2059,6 +2059,8 @@ void SceneTreeDock::_selection_changed() {
 		_tool_selected(TOOL_MULTI_EDIT);
 	} else if (selection_size == 0) {
 		editor->push_item(nullptr);
+	} else {
+		editor->push_item(EditorNode::get_singleton()->get_editor_selection()->get_selected_node_list()[0]);
```
Comment (akien-mga) [inline editor/scene_tree_dock.cpp:2063]: I wonder if that's the proper fix. Currently changing selection from one node to another works fine, so it means this new `push_item` would be redundant when going from single selection to single selection.

Comment (unsignedFoo) [reply editor/scene_tree_dock.cpp:2063]: Ah I see. But how we can tell to the editor we are swapping? I think this need more rework than a simple couple of lines. Like... changing how editor for MultiNodeEdit works, no?

Comment (EricEzaM) [reply editor/scene_tree_dock.cpp:2063]: Seems fine to me, it is also used in CanvasItemEditor:

https://github.com/godotengine/godot/blob/a9c53fa5994db8badb4ec14581bd8422fb340f41/editor/plugins/canvas_item_editor_plugin.cpp#L726-L728

Comment (EricEzaM) [reply editor/scene_tree_dock.cpp:2063]: I think I found the fix, and submitted PR #49725. Don't mean to step on any toes - I just found some other bugs at the same time so tried to fix those as well.

See here for my solution to this issue:
https://github.com/godotengine/godot/pull/49725/files#diff-80799453dbf9f78a65dda9407ff4c117de273e621d485eca54703dffbcff2de1R662-R669

Comment (unsignedFoo) [reply editor/scene_tree_dock.cpp:2063]: I think then we can close mine. I like your approach, let's see what Akien says :)

**Pull request closed without merging**